libraries/ESP32/examples/GPIO/FunctionalInterrupt/FunctionalInterrupt.ino

```diff
@@ -1,47 +1,39 @@
 #include <Arduino.h>
-#include <FunctionalInterrupt.h>
 
 #define BUTTON1 16
 #define BUTTON2 17
 
-class Button
-{
-public:
-	Button(uint8_t reqPin) : PIN(reqPin){
-		pinMode(PIN, INPUT_PULLUP);
-		attachInterrupt(PIN, std::bind(&Button::isr,this), FALLING);
```

Comment:
While your changes provide simpler code, the original example shows how to attach an interrupt to a class method (and that is often asked), so I suggest to keep the old way and use `class` instead of `struct`

Reply:
@pedrominatel could you please explain why we went from c++ to c?

Reply:
It was just a matter of getting consistency across many examples in C, and not in C++.

I would prefer to keep the `FunctionalInterrupt` example as is and add a new one using a simpler code, as you already did.

Reply:
So we agree that the example with `class Button` should stay and a new one with `struct Button` should be added?

Reply:
I agree. Is that ok for you @PilnyTomas?

Reply:
@PilnyTomas this is still left to be done